asset/records.go

```diff
@@ -87,32 +87,74 @@ func AssertNoUnknownEvenTypes(parsedTypes tlv.TypeMap,
 	return nil
 }
 
+// FilterUnknownTypes filters out all types that are unknown from the given
+// parsed types. The known types are specified as a set.
+func FilterUnknownTypes(parsedTypes tlv.TypeMap,
+	knownTypes fn.Set[tlv.Type]) tlv.TypeMap {
+
+	result := make(tlv.TypeMap, len(parsedTypes))
```

Comment:
We should be able to use `Set.Diff` here: https://github.com/lightningnetwork/lnd/blob/b7c59b36a74975c4e710a02ea42959053735402e/fn/set.go#L34-L43

Reply:
Then we'd need to run through again to read the keys in the output of the diff from the `parsedTypes` argument. So LoC wise, the current version may actually be the most minimal.